Add support for multiple policy in on block event

asafambar opened the pull request:
- [ ] All [tests](https://github.com/jfrog/jfrog-cli-core#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [ ] All [static analysis checks](https://github.com/jfrog/jfrog-cli-core/actions/workflows/analysis.yml) passed.
- [ ] This pull request is on the dev branch.
- [ ] I used gofmt for formatting the code before submitting the pull request.
-----

eyalbe4 commented on utils/coreutils/tableutils.go:
```diff
@@ -91,6 +92,37 @@ var DefaultMaxColWidth = 25
+//	    CatNumber string   `col-name:"Product\nCatalog #" auto-merge:"true"`
+//	    Color     string   `col-name:"Color" extended:"true" auto-merge:"true"`
+//	}
+//  customersSlice := []Customer{
```
Let's add a new-line between the two structs.

eyalbe4 commented on utils/coreutils/tableutils.go:
```diff
@@ -140,6 +172,7 @@ func PrepareTable(rows interface{}, emptyTableMessage string, printExtended bool
 		columnName, columnNameExist := field.Tag.Lookup("col-name")
 		embedTable, embedTableExist := field.Tag.Lookup("embed-table")
 		extended, extendedExist := field.Tag.Lookup("extended")
+		_, autoMergeExist := field.Tag.Lookup("auto-merge")
```
autoMergeExist --> autoMerge

eyalbe4 commented on xray/commands/curation/audit.go:
```diff
@@ -279,24 +277,34 @@ func printResult(format utils.OutputFormat, projectPath string, packagesStatus [
 	var pkgStatusTable []PackageStatusTable
-	for _, pkgStatus := range packagesStatus {
+	for index, pkgStatus := range packagesStatus {
+		// We use auto-merge supported by "go-pretty" library. It doesn't have an option to merge lines by group of uniq fields.
+		// we make each group to be merged only with itself by adding/not adding space, this way it won't be merged with the next group.
```
Since thw above comment isn't perfect in terms of English grammar, let's run it using a brammar corerction tool.
For example - 
... supported by "go-pretty" library... --> ... supported by the "go-pretty" library...

eyalbe4 reviewed (changes requested):
Will you be able to add a clear description to this PR, explaining what this feature is all about and what `auto-merge` means?
I'd be happy ro review this PR again afterwards.

asafambar commented:
Description fo PR:
I added the support to present many policies which blocked the package instead of only one,
also added for each policy the explanation and recommendation,.
explanation - why the package was blocked by this policy (e.g: CVE2020 with severity 9.0)
recommendation - what to do in order to fix the issue (e.g: upgrade to 2.0.1 version)

In order to show multi policies in a neat way in the table structure, I added the `auto-merge` option,
this option is supported by the library we are using today ---> https://github.com/jedib0t/go-pretty
it merges cells which have similar values, so if under the same blocked package I have many policies with recommendations and explanations, it will merge the the lines for the same package under one, and will keep the split lines for the policies.
an output example:
![image](https://github.com/jfrog/jfrog-cli-core/assets/15627787/92d94d59-5eba-4683-96ac-0b1ec84b5ac3)

yahavi commented on utils/coreutils/tableutils.go:
```diff
@@ -91,6 +92,38 @@ var DefaultMaxColWidth = 25
+// ├──────┼─────┼───────────────────────────┼───────────┤
+// │ Noah │ 21  │ Pouch                     │ 456789    │
+// └──────┴─────┴───────────────────────────┴───────────┘
+//
+//
```
```suggestion
```

yahavi commented on xray/commands/curation/audit.go:
```diff
@@ -74,25 +74,23 @@ type PackageStatus struct {
 }
 
 type Policy struct {
```
Can we add "omitempty" to all JSON fields in the structs in this file?

asafambar replied:
added to some.

yahavi commented on xray/commands/curation/audit.go:
```diff
@@ -463,6 +471,14 @@ func (nc *treeAnalyzer) extractPoliciesFromMsg(respError *ErrorsResp) []Policy {
+		if len(polCond) == 4 {
+			pol := polCond[0]
+			cond := polCond[1]
+			exp := strings.ReplaceAll(strings.Replace(polCond[2], ": ", ":\n", 1), " | ", "\n")
+			rec := strings.ReplaceAll(strings.Replace(polCond[3], ": ", ":\n", 1), " | ", "\n")
```
It is not clear what this replacement does. Let's export it to a private function to be more clear and remove the code duplication.

yahavi commented on xray/commands/curation/audit.go:
```diff
@@ -463,6 +471,14 @@ func (nc *treeAnalyzer) extractPoliciesFromMsg(respError *ErrorsResp) []Policy {
 			cond := polCond[1]
 			policies = append(policies, Policy{Policy: strings.TrimSpace(pol), Condition: strings.TrimSpace(cond)})
 		}
+		if len(polCond) == 4 {
+			pol := polCond[0]
+			cond := polCond[1]
```
Is it possible to have 0, 1, or 3 columns? If not, let's put the pol and col outside of the if's, to remove code duplication.
```suggestion
	        pol := polCond[0]
		cond := polCond[1]
		if len(polCond) == 4 {
```

asafambar commented on xray/commands/curation/audit.go:
```diff
@@ -450,23 +458,37 @@ func (nc *treeAnalyzer) getBlockedPackageDetails(packageUrl string, name string,
 }
 
+// Adding a new line after the headline and replace every "|" with a new line.
+func makeLegiblePolicyDetails(explanation, recommendation string) (string, string) {
```
I have unit test for the function which using it, anyway, I can add dedicated one
